### PR TITLE
SNOW-1759821 - [GitHub] Too many Warnings: "might have an offset token mismatch based on the provided offset token verification logic"

### PR DESCRIPTION
**Pull request by sfc-gh-achyzy**
# Overview

SNOW-1759821 [GitHub] Too many Warnings: "might have an offset token mismatch based on the provided offset token verification logic"

Original issue: #961 

I changed the logic to only validate if the offsets are incremental - not that they differ by 1. 

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [x] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

**Review by sfc-gh-dseweryn — APPROVED** [review]

**Review by sfc-gh-gjachimko — APPROVED** [review]
LGTM!

**Review by sfc-gh-dseweryn — APPROVED** [review]
👍

File: src/test/java/com/snowflake/kafka/connector/ConnectorConfigValidatorTest.java [hunk 1]
```diff
@@ -838,6 +838,53 @@ public void testEnableStreamingChannelMigrationConfig_invalidWithSnowpipe() {
             SnowflakeSinkConnectorConfig.ENABLE_CHANNEL_OFFSET_TOKEN_MIGRATION_CONFIG);
   }
 
+  @Test
+  public void testEnableStreamingChannelOffsetVerificationConfig() {
+    Map<String, String> config = getConfig();
```
Comment (sfc-gh-mbobowski) [inline src/test/java/com/snowflake/kafka/connector/ConnectorConfigValidatorTest.java:843]: I have recently added a proper builder for configs. Please use `SnowflakeSinkConnectorConfigBuilder.snowpipeConfig().build()` instead.

Comment (sfc-gh-mbobowski) [reply src/test/java/com/snowflake/kafka/connector/ConnectorConfigValidatorTest.java]: Fixed with Artur's approval.

**Review by sfc-gh-mbobowski — APPROVED** [review]

**Pull request merged**